svelte/src/lib/components/dropdown/Dropdown.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import type { Snippet } from 'svelte';
+  import type { HTMLAttributes } from 'svelte/elements';
+  import type { DropdownContext } from './context';
+
+  import { clickOutside } from './clickOutside';
+  import { setDropdown } from './context';
+
+  interface Props extends HTMLAttributes<HTMLDivElement> {
+    children: Snippet;
+  }
+
+  let { children, class: className, ...restProps }: Props = $props();
+
+  let isExpanded = $state(false);
+  let triggerId = `dropdown-trigger-${crypto.randomUUID().slice(0, 8)}`;
+  let contentId = `dropdown-content-${crypto.randomUUID().slice(0, 8)}`;
+
+  let context: DropdownContext = {
+    get isExpanded() {
+      return isExpanded;
+    },
+    triggerId,
+    contentId,
+    toggle: () => (isExpanded = !isExpanded),
+    close: () => (isExpanded = false),
+  };
```

Comment:
use [`$props.id()`](https://svelte.dev/docs/svelte/$props#$props.id()) instead.

```suggestion
  let id = $props.id();
  let isExpanded = $state(false);
  let triggerId = $derived(`dropdown-trigger-${id}`);
  let contentId = $derived`dropdown-content-${id}`);

  let context: DropdownContext = {
    get isExpanded() {
      return isExpanded;
    },
    get triggerId() {
      return triggerId
    },
    get contentId() {
      return contentId
    },
    toggle: () => (isExpanded = !isExpanded),
    close: () => (isExpanded = false),
  };
```